src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -1701,49 +1701,51 @@ private void ConvertProjectReferencesToMetadataReferences_NoLock(ProjectId proje
 
         public ProjectReference? TryCreateConvertedProjectReference(ProjectId referencingProject, string path, MetadataReferenceProperties properties)
```

Comment:
should this have the name _NoLock?

Reply:
Yeah you're right. At first it felt a bit funky since it's an internal-ish method but better to call out the evil than be surprised by it later.